src/main/java/com/mastery/java/task/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,190 @@
+package com.mastery.java.task.dao;
+
+import com.mastery.java.task.dto.Employee;
+import com.mastery.java.task.dto.Gender;
+import org.springframework.stereotype.Component;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class EmployeeDaoImpl implements EmployeeDao {
+
+    private final DataSource dataSource;
+
+    public EmployeeDaoImpl(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+
+    @Override
+    public void save(Employee employee) {
+        String query = "INSERT INTO employee (first_name, last_name, departament_id, job_title, gender, date_of_birth) values (?, ?, ?, ?, ?, ?)";
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        try {
+            connection = dataSource.getConnection();
+            preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, employee.getFirstName());
+            preparedStatement.setString(2, employee.getLastName());
+            preparedStatement.setInt(3, employee.getDepartamentId());
+            preparedStatement.setString(4, employee.getJobTitle());
+            preparedStatement.setString(5, String.valueOf(employee.getGender()));
+            preparedStatement.setDate(6, (Date) employee.getDateOfBirth());
+
+            int out = preparedStatement.executeUpdate();
+            if (out != 0) {
+                System.out.println("Employee saved with id=" + employee.getEmployeeId());
+            } else System.out.println("Employee save failed with id=" + employee.getEmployeeId());
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                preparedStatement.close();
+                connection.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+
+    public Employee getById(int id) {
+
+        String query = "SELECT employee_id, first_name, last_name, departament_id, job_title, gender, date_of_birth from employee where employee_id = ?";
```

Comment:
Please use named constant String in the top of the class to store SQL query

Reply:
Done